Code-the-Dream-School/cc-prac-team4-front
# Pp 112 add functionality to search on pats page

**Functionality to search on pats page implemented**

**Pull Request Description:**

**This pull request includes changes to the codebase related to searching pet names. Here are the modifications made:**

- In PetActions.js, added a new asynchronous function getSearchPetName that handles the search for a pet name. It dispatches actions to update the state during the search process.

- In PetReducer.js, added a new reducer function SearchPetNameReducer that handles different actions related to searching pet names. It updates the state accordingly based on the dispatched actions.

- In Header.js, added an useEffect hook that listens for changes in the petNameResponse state and saves it to the local storage. It also sends a message to other components when the petNameResponse is updated. Also minor changes was added to this file related to the search functionality. The search form now handles the search for pet names.

- Pets.js sets up an useEffect hook that listens for changes in the local storage. When the petNameResults key is updated in the local storage, the code retrieves the updated value, parses it as JSON, and sets the pets state with the parsed results. This allows the component to respond to changes in the petNameResults stored in the local storage and update the UI accordingly.

**Testing Steps:**

1. Use the search form in the header to search for a pet name.
2. Verify that the search results are displayed correctly.
3. Perform additional tests to ensure the correct behavior of the search functionality.
4. Verify that the state updates are being saved to the local storage and messages are sent when the petNameResponse is updated.
5. Test other areas of the application to ensure that these changes do not introduce any regressions.
6. Verify that the application functions as expected and there are no console errors or unexpected behaviors.

<img width="1422" alt="Screenshot 2023-07-04 at 12 01 16 AM" src="https://github.com/Code-the-Dream-School/cc-prac-team4-front/assets/2952900/6bdbc867-0f53-4f7c-98d1-b78548e95cb6">
